lms/events/__init__.py

```diff
@@ -0,0 +1,70 @@
+from dataclasses import dataclass
+
+from pyramid.events import subscriber
+from pyramid.request import Request
+
+from lms.models import Event, EventType, LtiLaunches
+from lms.services import EventService, LTIRoleService
+
+
+@dataclass
+class LTILaunchEvent:
+    request: Request
+
+
+@dataclass
+class LTIDeepLinkingEvent:
+    request: Request
+    document_url: str
+
+
+@subscriber(LTILaunchEvent)
+def handle_lti_launch_event(event: LTILaunchEvent):
+    request = event.request
+
+    # Record Launch in the legacy LtiLaunches table
+    LtiLaunches.add(
+        request.db,
+        request.lti_params.get("context_id"),
+        request.lti_params.get("oauth_consumer_key"),
+    )
+
+    # Record the launch as en event in the Event model's table
+    request.find_service(EventService).insert_event(
+        type_=EventType.Type.CONFIGURED_LAUNCH,
+        user=request.user,
```

Comment:
We could potentially load the event up with some of this information but keeping the event very thin also works.